### PR TITLE
Return pod metadata in pod list and status APIs

### LINKED ISSUE #107: Add pod `--name` to ocic pod list?
What's the purpose of --name on the ocic create command?

I get that the id is automatically generated. But I figured if we're going to have a requested name it should at least show up on list & status.

We are not currently passing metadata back from the list command nor storing it?

```
type PodSandbox struct {
	// The id of the PodSandbox
	Id *string `protobuf:"bytes,1,opt,name=id" json:"id,omitempty"`
	// Metadata of the sandbox
	Metadata *PodSandboxMetadata `protobuf:"bytes,2,opt,name=metadata" json:"metadata,omitempty"`
	// The state of the PodSandbox
	State *PodSandBoxState `protobuf:"varint,3,opt,name=state,enum=runtime.PodSandBoxState" json:"state,omitempty"`
	// Creation timestamps of the sandbox
	CreatedAt *int64 `protobuf:"varint,4,opt,name=created_at,json=createdAt" json:"created_at,omitempty"`
	// The labels of the PodSandbox
	Labels           map[string]string `protobuf:"bytes,5,rep,name=labels" json:"labels,omitempty" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
	XXX_unrecognized []byte            `json:"-"`
}

// PodSandboxStatus contains the status of the PodSandbox.
type PodSandboxStatus struct {
	// ID of the sandbox.
	Id *string `protobuf:"bytes,1,opt,name=id" json:"id,omitempty"`
	// Metadata of the sandbox.
	Metadata *PodSandboxMetadata `protobuf:"bytes,2,opt,name=metadata" json:"metadata,omitempty"`
	// State of the sandbox.
	State *PodSandBoxState `protobuf:"varint,3,opt,name=state,enum=runtime.PodSandBoxState" json:"state,omitempty"`
	// Creation timestamp of the sandbox
	CreatedAt *int64 `protobuf:"varint,4,opt,name=created_at,json=createdAt" json:"created_at,omitempty"`
	// Network contains network status if network is handled by the runtime.
	Network *PodSandboxNetworkStatus `protobuf:"bytes,5,opt,name=network" json:"network,omitempty"`
	// Linux specific status to a pod sandbox.
	Linux *LinuxPodSandboxStatus `protobuf:"bytes,6,opt,name=linux" json:"linux,omitempty"`
	// Labels are key value pairs that may be used to scope and select individual resources.
	Labels map[string]string `protobuf:"bytes,7,rep,name=labels" json:"labels,omitempty" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
	// Annotations is an unstructured key value map that may be set by external
	// tools to store and retrieve arbitrary metadata.
	Annotations      map[string]string `protobuf:"bytes,8,rep,name=annotations" json:"annotations,omitempty" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
	XXX_unrecognized []byte            `json:"-"`
}

server/sandbox.go:

type sandbox struct {
	id          string
	name        string
	logDir      string
	labels      map[string]string
	annotations map[string]string
	containers  oci.Store
}
```

### PR DESCRIPTION
**Pull request by mrunalp**
Close #107 

@mikebrow PTAL

(This could use some additional tests in a follow up).

### CODE REVIEW

**Comment by runcom** [comment]
LGTM

File: cmd/client/sandbox.go [hunk 1]
```diff
@@ -242,6 +268,32 @@ func ListPodSandboxes(client pb.RuntimeServiceClient) error {
 		fmt.Printf("Status: %s\n", pod.State)
 		ctm := time.Unix(*pod.CreatedAt, 0)
 		fmt.Printf("Created: %v\n", ctm)
+		if pod.Metadata != nil {
```
Comment (mikebrow) [inline cmd/client/sandbox.go]: nit.. status output sequence goes from id to metadata to status then created..
list output sequence goes from id to status to created then metadata.. 

Suggest picking one to use for both..

**Comment by mikebrow** [comment]
LGTM works great... just one minor nit.

File: cmd/client/sandbox.go (hunk 1 above)
Comment (mrunalp) [inline cmd/client/sandbox.go]: Fixed.

**Pull request merged**